ts/components/helpers/withMessagesSelection.tsx

```diff
@@ -1,16 +1,22 @@
-import { none, Option, some } from "fp-ts/lib/Option";
+import { isSome, none, Option, some } from "fp-ts/lib/Option";
```

Comment:
This HOC must be used just to maintain the list of selected messagesIds. It has been created to share the logic between multiple component and to avoid to create the same state and functions for each component. If you already know react "hooks" it is like a useMessagesSelection. It is wrapped/using component responsibility to know if the selectedMessagesIds correspond to all the messages the component show.

Reply:
@fpersico Changed the logic as suggested in the commit 3aba4b0